### PR TITLE
mrc-1529 validation when adding new regions

### PR DESCRIPTION
**Pull request by hillalex**
Prevents empty or duplicate region names from being added, both in the new header widget, and from the project list page.

Introduces slugs for projects and regions, essentially unique ids which allow us to identify the project/region from any given url, which will come in handy for the next ticket around makes routes properly navigable.

### CODE REVIEW

File: src/app/static/src/app/components/app.vue [hunk 1]
```diff
@@ -77,7 +82,19 @@
         },
         components: {dropDown, BIconGraphUp, BModal},
         directives: {"BModal": VBModal},
-        computed: mapState(["currentProject", "baselineOptions", "interventionOptions"]),
+        computed: {
+            ...mapState(["currentProject", "baselineOptions", "interventionOptions"]),
+            validNewRegion() {
+                if (this.newRegionName.length == 0) {
```
Comment (EmmaLRussell) [inline src/app/static/src/app/components/app.vue:88]: Could trim to check for whitespace here too

**Review by EmmaLRussell — COMMENTED** [review]
Would be nice to have a bit of user feedback, for duplicates especially

File: src/app/static/src/app/components/projectListPage.vue [hunk 2]
```diff
@@ -118,24 +144,48 @@
             },
             createProject() {
                 const regionNames = this.regions.map((tag) => tag.text.trim());
-                if (regionNames.length == 0) {
-                    // user has only entered one region name and has not blurred the input
-                    // so this.regions is empty even though this.newRegion is populated
-                    // so take this.newRegion as the only region
-                    regionNames.push(this.newRegion)
+                const newRegion = this.newRegion.trim();
+                if (newRegion.length > 0) {
```
Comment (hillalex) [inline src/app/static/src/app/components/projectListPage.vue:148]: Have also changed this so it always adds the last typed region (once validated) even if the user hasn't blurred the input and created a tag.

Comment (EmmaLRussell) [reply src/app/static/src/app/components/projectListPage.vue:148]: Good idea, that's caught me out before!

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**